tools/bug-report/pkg/kubectlcmd/kubectlcmd.go

```diff
@@ -55,16 +90,40 @@ func Logs(client kube.ExtendedClient, namespace, pod, container string, previous
 	if dryRun {
 		return fmt.Sprintf("Dry run: would be running client.PodLogs(%s, %s, %s)", pod, namespace, container), nil
 	}
+	requestLimiter.Wait(context.TODO())
+	logFetchLimitCh <- struct{}{}
+	defer func() {
+		<-logFetchLimitCh
+	}()
+	task := fmt.Sprintf("PodLogs %s/%s/%s", namespace, pod, container)
+	addRunningTask(task)
+	defer removeRunningTask(task)
 	return client.PodLogs(context.TODO(), pod, namespace, container, previous)
 }
 
+// EnvoyGet sends a GET request for the URL in the Envoy container in the given namespace/pod and returns the result.
+func EnvoyGet(client kube.ExtendedClient, namespace, pod, url string, dryRun bool) (string, error) {
+	if dryRun {
+		return fmt.Sprintf("Dry run: would be running client.EnvoyDo(%s, %s, %s)", pod, namespace, url), nil
+	}
+	requestLimiter.Wait(context.TODO())
+	task := fmt.Sprintf("ProxyGet %s/%s:%s", namespace, pod, url)
+	addRunningTask(task)
+	defer removeRunningTask(task)
+	out, err := client.EnvoyDo(context.TODO(), pod, namespace, "GET", url, nil)
+	return string(out), err
+}
+
 // Cat runs the cat command for the given path in the given namespace/pod/container.
 func Cat(client kube.ExtendedClient, namespace, pod, container, path string, dryRun bool) (string, error) {
 	cmdStr := "cat " + path
 	if dryRun {
 		return fmt.Sprintf("Dry run: would be running podExec %s/%s/%s:%s", pod, namespace, container, cmdStr), nil
 	}
-	log.Infof("podExec %s/%s/%s:%s", namespace, pod, container, cmdStr)
+	requestLimiter.Wait(context.TODO())
```

Comment:
Is this limiting concurrency or rate?
I think we want to limit concurrency here right? 10 concurrent log fetches would send significant amounts of data thru api server.

Reply:
done.